machine-runner/tests/esm/runner.test.ts

```diff
@@ -308,6 +309,33 @@ describe('machine runner', () => {
     r.machine.destroy()
     r.assertSubscribed(false)
   })
+
+  describe('zod-support', () => {
+    const { On, Off, Events } = ProtocolThreeSwitch
+
+    it('should not allow payload with a wrong schema', () => {
+      expect(() => Events.ToggleOn.make({ value: 1 } as any)).toThrow() // wrong payload schema
```

Comment:
Why does this line compile?

Reply:
Related but different question: is it possible to write a zod schema whose TypeScript type accepts values that the schema does not?

Reply:
> Why does this line compile?

`as any`, it breaches everything.

> is it possible to write a zod schema whose TypeScript type accepts values that the schema does not?

Yes. If the zod type has a [refinement](https://zod.dev/?id=refine). Refinement plays in the realm of JS values.
In our case:
- `MachineEvent.make` will throw invalid values,
- `MachineRunnerInternals.pushEvent` will mark invalid values as event that does not match the factory

Reply:
Okay, that matches my earlier expectation. I must have overlooked the code that throws in `make`, though, and I think it would be good to have two tests, one that asserts the compile-time error and one with a refinement.

Reply:
done